Client/Frontend/Home/HomepageViewController.swift

```diff
@@ -334,8 +334,10 @@ extension HomepageViewController: UICollectionViewDelegate, UICollectionViewData
         }
         
         // Ecosia: footer for impact
-        if sectionViewModel.sectionType == .impact, kind == UICollectionView.elementKindSectionFooter {
-            return collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: NTPImpactDividerFooter.cellIdentifier, for: indexPath)
+        if sectionViewModel.sectionType == .impact, kind == UICollectionView.elementKindSectionFooter,
+           let footer = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: NTPImpactDividerFooter.cellIdentifier, for: indexPath) as? NTPImpactDividerFooter {
+            footer.applyTheme()
```

Comment:
How about making use of the existing `DisplayThemeChanged` notification and add an observer for `applyTheme` inside `NTPImpactDividerFooter` so that no external code to the class is needed?

I've noticed this logic for applying themes being used in other places, for example the `NTPTooltip`.

Reply:
Nice one @lucaschifino! I did check how the observer is implemented in the `NTPTooltip` and I wonder whether a similar implementation could introduce a memory leak.
The observer in the `NTPTooltip` is added upon initialization as well and the function `applyTheme()` is executed.
As the view is a `UICollectionReusableView` we have no control over its instance (which is fine), however, it seems like the observer still lives (forcing the whole view to stay alive) even though the view could be deallocated.
If this flow resonates with you, are you happy with applying your suggestion as well as a `deinit()` that removes the observer?

Reply:
Yes, you're right! Very good addition with the `deinit` 😁

Reply:
Cool, will apply the `deinit()` to both the `NTPTooltip` and the `NTPImpactDividerFooter` if you're happy with that, even though it's not 💯 the same scope.

Reply:
Edit: I had a read at the Apple Doc as I questioned myself on the why a `deinit()` was not preset in more than a place and found out that a `deinit()` is indeed needed, but since newer iOS versions, only for block-based observers.

> If the observer is able to be stored as a zeroing-weak reference the underlying storage will store the observer as a zeroing weak reference, alternatively if the object cannot be stored weakly (i.e. it has a custom retain/release mechanism that would prevent the runtime from being able to store the object weakly) it will store the object as a non-weak zeroing reference. This means that observers are not required to un-register in their deallocation method.
Block based observers via the -[NSNotificationCenter addObserverForName: object: queue: usingBlock] method still need to be un-registered when no longer in use since the system still holds a strong reference to these observers.